app/models/note.rb

```diff
@@ -0,0 +1,8 @@
+class Note < Mongoid_Base
+  belongs_to :document
+
+  has_many :tag_arrays
```

Comment:
Shouldn't note have many tag_arrays with each tag_array containing each different reference of the tag for each document and note that uses the tag? So if note can have multiple tags than it must have multiple tag arrays since each tag array is really only one tag.